packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +171,317 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+      setCaretIndex(selectionEnd + displayName.length);
+      setCurrentTriggerStartIndex(-1);
+      updateMentionSuggestions([]);
+      setActiveSuggestionIndex(undefined);
+      onChange && onChange(null, updatedHTML);
+    },
+    [
+      textFieldRef,
+      inputTextValue,
+      currentTriggerStartIndex,
+      mentionLookupOptions?.trigger,
+      onChange,
+      textValue,
+      tagsValue,
+      /* @conditional-compile-remove(mention) */
+      updateMentionSuggestions,
+      /* @conditional-compile-remove(mention) */
+      localeStrings
+    ]
+  );
+
+  const onTextFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
       // Uses KeyCode 229 and which code 229 to determine if the press of the enter key is from a composition session or not (Safari only)
       if (ev.nativeEvent.isComposing || ev.nativeEvent.keyCode === 229 || ev.nativeEvent.which === 229) {
         return;
       }
+      if (ev.key === 'ArrowUp') {
+        ev.preventDefault();
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? mentionSuggestions.length - 1
+              : Math.max(activeSuggestionIndex - 1, 0);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      } else if (ev.key === 'ArrowDown') {
+        ev.preventDefault();
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? 0
+              : Math.min(activeSuggestionIndex + 1, mentionSuggestions.length - 1);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      }
       if (ev.key === 'Enter' && (ev.shiftKey === false || !supportNewline)) {
         ev.preventDefault();
+
+        // If we are looking up a mention, select the focused suggestion
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0 && activeSuggestionIndex !== undefined) {
+          const selectedMention = mentionSuggestions[activeSuggestionIndex];
+          if (selectedMention) {
+            onSuggestionSelected(selectedMention);
+            return;
+          }
+        }
+
         onEnterKeyDown && onEnterKeyDown();
       }
       onKeyDown && onKeyDown(ev);
     },
-    [onEnterKeyDown, onKeyDown, supportNewline]
+    [
+      onEnterKeyDown,
+      onKeyDown,
+      supportNewline,
+      /* @conditional-compile-remove(mention) */
+      mentionSuggestions,
+      /* @conditional-compile-remove(mention) */
+      activeSuggestionIndex,
+      /* @conditional-compile-remove(mention) */
+      onSuggestionSelected
+    ]
   );
 
+  /* @conditional-compile-remove(mention) */
+  const debouncedQueryUpdate = useRef(
+    useDebouncedCallback(async (query: string) => {
```

Comment:
Why do we useRef then .current here?

Reply:
Is it just call the hook for once? Ideally react component wouldn't allow you to conditional call a hook just for once

Reply:
I guess useMeme should be working well if that is the case